### ISSUE TITLE
ESLintUtils.getParserServices should take a Readonly<RuleContext>

### ISSUE THREAD

**Issue by danielnixon** [typescript-eslint/typescript-eslint#2232]
**Repro**

I'm writing [an ESLint rule](https://github.com/danielnixon/eslint-plugin-total-functions/blob/master/src/rules/no-unsafe-assignment.ts#L13) to ban code that leads to the surprising mutability described in [TypeScript issue #13347](https://github.com/microsoft/TypeScript/issues/13347).

I'd like to forbid passing an immutable value to a function that expects a mutable (but otherwise compatible) type.

I've noticed that `RuleModule.create` gives us an `Readonly<RuleContext>` but `ESLintUtils.getParserServices` expects a (mutable) `RuleContext`. This triggers my rule (see below).

It looks like `ESLintUtils.getParserServices` doesn't mutate the passed in `context`, so please consider updating it to take a `Readonly<RuleContext>`. This would stop my rule from complaining.

```TS
const rule: RuleModule<...> = {
  meta: {
      ...
  },
  create: (context) => { // Context is a Readonly<RuleContext>
    const parserServices = ESLintUtils.getParserServices(context); // but getParserServices takes a (mutable) RuleContext
```

**Expected Result**

`ESLintUtils.getParserServices` takes `Readonly<RuleContext>`

**Actual Result**

`ESLintUtils.getParserServices` takes (mutable) `RuleContext`

**Additional Info**

* https://github.com/danielnixon/eslint-plugin-total-functions/issues/21
* https://github.com/danielnixon/eslint-plugin-total-functions
* https://github.com/microsoft/TypeScript/issues/13347
* https://github.com/microsoft/TypeScript/issues/14150

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `3.3.0` and earlier |
| `TypeScript`                            | NA |
| `node`                                  | NA |
| `npm`                                   | NA |

**Comment by bradzacher** [comment]
Happy to accept a PR!
